shared/pkg/analyzer/syft/syft.go

```diff
@@ -51,10 +52,12 @@ func New(conf *config.Config,
 		logger:     logger.Dup().WithField("analyzer", AnalyzerName),
 		config:     config.CreateSyftConfig(conf.Analyzer, conf.Registry),
 		resultChan: resultChan,
+		localImage: conf.LocalImageScan,
 	}
 }
 
-func (a *Analyzer) Run(sourceType utils.SourceType, src string) error {
+func (a *Analyzer) Run(sourceType utils.SourceType, userInput string) error {
+	src := utils.SetSource(a.localImage, sourceType, userInput)
 	src = utils.CreateUserInput(sourceType, src)
```

Comment:
it looks like `utils.CreateUserInput` and `utils.SetSource` should be combined into a single function, am I wrong?
What is the expected input in `CreateResult`? The source? The modified user input? The original user input? We need to better understand what is used where in both syft and grype.

Reply:
You're right, It can be one function. In the case of the input type is an image we have to extend the image soruce with docker or registry for syft and grype. I will update it.

Reply:
The expected input should be the `userInput` in the case of `CreateResults`.